packages/fx-core/src/plugins/resource/sql/utils/telemetryUtils.ts

```diff
@@ -42,14 +39,17 @@ export class TelemetryUtils {
     properties[Telemetry.properties.errorCode] = errorCode;
     properties[Telemetry.properties.errorType] = errorType;
     properties[Telemetry.properties.errorMessage] = errorMessage;
-    properties[Telemetry.properties.component] = Telemetry.componentName;
-    if (this.ctx.app.id) {
-      properties[Telemetry.properties.appid] = this.ctx.app.id;
-    }
+    this.addProperties(properties);
     TelemetryUtils.ctx.telemetryReporter?.sendTelemetryErrorEvent(
       eventName,
       properties,
       measurements
     );
   }
+
+  private static addProperties(properties: { [key: string]: string; }) {
```

Comment:
Seems there are duplicate code between sql telemetryUtils and identity telemetryUtils, is this by design?

Reply:
Yes, currently we don't share code across plugins;